beacon/goclient/proposer.go

```diff
@@ -187,17 +189,85 @@ func (gc *GoClient) SubmitBlindedBeaconBlock(block *api.VersionedBlindedProposal
 		Proposal: signedBlock,
 	}
 
-	start := time.Now()
-	err := gc.client.SubmitBlindedProposal(gc.ctx, opts)
-	recordRequestDuration(gc.ctx, "SubmitBlindedProposal", gc.client.Address(), http.MethodPost, time.Since(start), err)
-	if err != nil {
-		gc.log.Error(clResponseErrMsg,
-			zap.String("api", "SubmitBlindedProposal"),
-			zap.Error(err),
-		)
+	// As gc.multiClient doesn't have a method for blinded block submission
+	// (because it must be submitted to the same node that returned that block),
+	// we need to submit it to client(s) directly.
+	if len(gc.clients) == 1 {
+		start := time.Now()
+		err := gc.clients[0].SubmitBlindedProposal(gc.ctx, opts)
+		recordRequestDuration(gc.ctx, "SubmitBlindedProposal", gc.clients[0].Address(), http.MethodPost, time.Since(start), err)
+		if err != nil {
+			gc.log.Error(clResponseErrMsg,
+				zap.String("api", "SubmitBlindedProposal"),
+				zap.Error(err),
+			)
+			return err
+		}
+
+		return nil
 	}
 
-	return err
+	// Although we got a blinded block from one node and that node has to submit it,
+	// other nodes might know this payload too and have a chance to submit it successfully.
+	//
+	// So we do the following:
+	//
+	// Submit the blinded proposal to all clients concurrently.
+	// If any client succeeds, cancel the remaining submissions.
+	// Wait for all submissions to finish or timeout after 1 minute.
+	//
+	// TODO: Make sure this the above is correct. Should we submit only to the node that returned the block?
```

Comment:
we should test this code thoroughly. 
is it correct that other nodes might know the payload for a blinded block though?